prow/plugins/override/override.go

```diff
@@ -64,49 +65,61 @@ type overrideClient interface {
 	githubClient
 	prowJobClient
 	ownersClient
-	presubmitForContext(org, repo, context string) *config.Presubmit
+	presubmits(org, repo string, baseSHAGetter config.RefGetter, headSHA string) ([]config.Presubmit, error)
 }
 
 type client struct {
-	gc            githubClient
-	jc            config.JobConfig
+	ghc           githubClient
+	gc            *git.Client
+	config        *config.Config
 	ownersClient  ownersClient
 	prowJobClient prowJobClient
```

Comment:
Why is this thing btw creating a ProwJob CR instead of just doing an api call to GitHub? At a quick glance this looks to me like an overly complicated way to achieve the result.

Reply:
Otherwise `tide` won't honor it, it reads ProwJobs